python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -962,6 +962,9 @@ def date_range(
         elif periods == 0:
             # end == start, return exactly 1 timestamp (start)
             periods = 1
+        else:
+            # Since closed="both", ensure end point is included
+            periods += 1
```

Comment:
Just realized this after approving. Can these be combined?
```suggestion
        else:
            # If end == start, periods == 0 and we return exactly 1 timestamp (start).
            # Otherwise, since closed="both", we ensure the end point is included.
            periods += 1
```

Reply:
Yup, this looks valid. Thanks!